### DISCUSSION TITLE
Install in venv on python 3.11

### DISCUSSION THREAD

**Discussion by linsui** [pypa/setuptools#4284] [Q&A] [answered]
On python 3.9 install and install_headers work well in venv. Files are installed locally. On 3.11 it tried to install files into global path. What is changed? How can I fix the problem? Thanks for any help!

**Comment by webknjaz** [comment]
If you didn't use pip, that's unsupported.

**Reply by linsui** [reply]
Is the function removed? How can I do that with pip?

**Reply by webknjaz** [reply]
What function?

**Reply by linsui** [reply]
The setup.py can install files to the venv path by default on python 3.9. Can I get the same behavior on python 3.11?

**Reply by webknjaz** [reply, marked as answer]
https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html

**Reply by linsui** [reply]
Oh, now I understand what you mean. Thanks!
